test/Microsoft.Extensions.Primitives.Tests/StringSegmentTest.cs

```diff
@@ -92,15 +92,17 @@ public void StringSegmentConstructor_NullBuffer_Throws()
         public void StringSegmentConstructor_NegativeOffset_Throws()
         {
             // Arrange, Act and Assert
-            var exception = Assert.Throws<ArgumentOutOfRangeException>(() => new StringSegment("", -1, 0));
+            // offset + length = 0 for the test
+            var exception = Assert.Throws<ArgumentOutOfRangeException>(() => new StringSegment("lengthof9", -1, 1));
```

Comment:
Why the change?

Reply:
To be honest I can't remember the reason for the change, but I must have had some motivation to do it.
Reconstructing my thoughts I believe that while optimizing I tried a variant for the ctor-arg-check with
```c#
if (buffer == null || (uint)offset > (uint)buffer.Length || (uint)(length + offset) > (uint)buffer.Length || length < 0)
```
(or similar, substituting the subtraction with an additional comparison), so test got changed to catch this.
This try was not better, so the arg-check wasn't changed and this change in the test remained as artefact.

The Ctor-test is not really affected by this change, nor is the coverage better.
I'll revert this line, and the same in the test for `StringSegmentConstructor_NegativeLength_Throws`